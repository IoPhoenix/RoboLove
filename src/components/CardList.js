import React from 'react';
import Card from './Card';

const CardList = (props) => {
    
    const cardsArray = props.robots.map(user => {
        return (
            <Card
             key={user.id} 
             id={user.id}
             name={user.name}
             email={user.email}
            />
        );
    });

    return (
        <div className='flex flex-column flex-wrap justify-center items-center flex-row-ns'>
            {cardsArray}
        </div>
    );
}

export default CardList;
