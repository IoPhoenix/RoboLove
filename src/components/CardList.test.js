import { shallow, mount, render } from 'enzyme';
import React from 'react';
import CardList from './CardList';

it('expect to render CardList component', () => {
    const mockRobots = [
        {
            id: 1,
            name: 'Olga',
            username: 'OlgaF',
            email: 'olga@mail.com'
        }
    ];

    expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
});