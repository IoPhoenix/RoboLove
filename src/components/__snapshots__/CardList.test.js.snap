// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
      Array [
        Object {
          "email": "olga@mail.com",
          "id": 1,
          "name": "Olga",
          "username": "OlgaF",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="olga@mail.com"
          id={1}
          name="Olga"
        />,
      ],
      "className": "flex flex-column flex-wrap justify-center items-center flex-row-ns",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "email": "olga@mail.com",
          "id": 1,
          "name": "Olga",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="olga@mail.com"
            id={1}
            name="Olga"
          />,
        ],
        "className": "flex flex-column flex-wrap justify-center items-center flex-row-ns",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "email": "olga@mail.com",
            "id": 1,
            "name": "Olga",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
